src/main/java/com/github/dockerjava/core/DefaultDockerClientConfig.java

```diff
@@ -0,0 +1,467 @@
+        public final Builder withDockerHost(String dockerHost) {
+            checkNotNull(dockerHost, "uri was not specified");
+            this.dockerHost = URI.create(dockerHost);
+            return this;
+        }
+
+        public final Builder withApiVersion(RemoteApiVersion apiVersion) {
+            this.apiVersion = apiVersion.getVersion();
+            return this;
+        }
+
+        public final Builder withApiVersion(String apiVersion) {
+            this.apiVersion = apiVersion;
+            return this;
+        }
+
+        public final Builder withRegistryUsername(String registryUsername) {
+            this.registryUsername = registryUsername;
+            return this;
+        }
+
+        public final Builder withRegistryPassword(String registryPassword) {
+            this.registryPassword = registryPassword;
+            return this;
+        }
+
+        public final Builder withRegistryEmail(String registryEmail) {
+            this.registryEmail = registryEmail;
+            return this;
+        }
+
+        public Builder withRegistryUrl(String registryUrl) {
+            this.registryUrl = registryUrl;
+            return this;
+        }
+
+        public final Builder withDockerCertPath(String dockerCertPath) {
+            this.dockerCertPath = dockerCertPath;
+            return this;
+        }
+
+        public final Builder withDockerConfig(String dockerConfig) {
+            this.dockerConfig = dockerConfig;
+            return this;
+        }
+
+        public final Builder withDockerTlsVerify(String dockerTlsVerify) {
+            if (dockerTlsVerify != null) {
+                String trimmed = dockerTlsVerify.trim();
+                this.dockerTlsVerify = "true".equalsIgnoreCase(trimmed) || "1".equals(trimmed);
+            } else {
+                this.dockerTlsVerify = false;
+            }
+            return this;
+        }
+
+        public final Builder withDockerTlsVerify(Boolean dockerTlsVerify) {
+            this.dockerTlsVerify = dockerTlsVerify;
+            return this;
+        }
+
+        /**
+         * Overrides the default {@link SSLContext} that is used when calling {@link Builder#withDockerTlsVerify(true)} and
+         * {@link Builder#withDockerCertPath(String)}. This way it is possible to pass a custom {@link SSLContext} to the resulting
+         * {@link DockerClientConfig} that may be created by other means than the local file system.
+         */
+        public final Builder withCustomSslContext(SSLContext customSslContext) {
+            this.customSslContext = customSslContext;
+            return this;
+        }
+
+        public DefaultDockerClientConfig build() {
+
+            SSLContext sslContext = null;
+
+            if (customSslContext == null) {
+                if (dockerTlsVerify) {
+                    dockerCertPath = checkDockerCertPath(dockerCertPath);
+                    sslContext = new LocalDirectorySSLConfig(dockerCertPath).getSSLContext();
```

Comment:
Is it possible to store SSLConfig in DefaultClientConfig instead context?
I need know from what dockerCertPath connection was build.
Use case: calling default dockerclientconfig builder that gather configs, env vars, properties and they i extract dockercertPath for creating special remote test connection.